exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -16,19 +16,24 @@ import (
 // the producer are dropped.
 type boundedMemoryQueue[T any] struct {
 	component.StartFunc
+	QueueCapacityLimiter[T]
 	items chan queueRequest[T]
 }
 
 // NewBoundedMemoryQueue constructs the new queue of specified capacity, and with an optional
 // callback for dropped items (e.g. useful to emit metrics).
-func NewBoundedMemoryQueue[T any](capacity int) Queue[T] {
+func NewBoundedMemoryQueue[T any](capacityLimiter QueueCapacityLimiter[T]) Queue[T] {
 	return &boundedMemoryQueue[T]{
-		items: make(chan queueRequest[T], capacity),
+		QueueCapacityLimiter: capacityLimiter,
+		items:                make(chan queueRequest[T], capacityLimiter.Capacity()),
 	}
 }
 
 // Offer is used by the producer to submit new item to the queue. Calling this method on a stopped queue will panic.
 func (q *boundedMemoryQueue[T]) Offer(ctx context.Context, req T) error {
+	if !q.QueueCapacityLimiter.claim(req) {
```

Comment:
We should have an unbounded channel then and rely only on the QueueCapacityLimiter for this.

Reply:
I updated it to use an unbounded channel and utilize the request based limiter 👍